views/default/forms/usersettings/save.php

```diff
@@ -5,10 +5,18 @@
  * Plugins should extend "forms/account/settings" to add to the settings.
  */
 
-$form_body = elgg_view("forms/account/settings", $vars);
+$user = elgg_get_page_owner_entity();
 
-$form_body .= '<div class="elgg-foot">';
-$form_body .= elgg_view('input/submit', array('value' => elgg_echo('save')));
-$form_body .= '</div>';
+$form_body = elgg_view('forms/account/settings', $vars);
+
+$submit = elgg_view('input/submit', ['value' => elgg_echo('save')]);
+$hidden = '';
```

Comment:
there is always the crazy situation where someone does an `elgg_view_form("usersettings/save")` when logged out... unfortunately we can't assume anything in views